lighthouse-core/gather/driver.js

```diff
@@ -673,21 +673,31 @@ class Driver {
     return this.sendCommand('Runtime.enable');
   }
 
-  beginEmulation(flags) {
+  beginEmulation(options) {
+    return Promise.resolve().then(_ => {
+      if (!options.flags.disableDeviceEmulation) {
+        return emulation.enableNexus5X(this);
+      }
+    }).then(_ => {
+      return this.setThrottling(options.flags, {useThrottling: true});
+    });
+  }
+
+  setThrottling(cliFlags, passConfig) {
     const emulations = [];
 
-    if (!flags.disableDeviceEmulation) {
-      emulations.push(emulation.enableNexus5X(this));
+    if (!passConfig.useThrottling) {
```

Comment:
this is one case I think having it nested would make it more readable, took me a bit to spot the early return (though that may be more diff noise and it's easier without the diff)

thoughts on the following?
```
if (useThrottling) {
  if (!disableNetwork) emulations.push()
  if (!disableCpu) emulations.push
} else {
  emulations.push(...)
}

return Promise.all()
```